.github/workflows/release.yml

```diff
@@ -1,35 +1,58 @@
-name: release
+name: 🕊 Release
 on:
-  release:
-    types: [published]
+  push:
+    branches:
+      - release
+      - "release-*"
+      - "!release-experimental"
+      - "!release-experimental-*"
+
+concurrency: ${{ github.workflow }}-${{ github.ref }}
+
+env:
+  CI: true
 
 jobs:
   release:
+    name: 🦋 Changesets Release
     if: github.repository == 'remix-run/react-router'
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v2
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
-      - name: Set up Node
-        uses: actions/setup-node@v2
+      - name: ⎔ Setup Node
+        uses: actions/setup-node@v3
         with:
-          node-version: 14
-          check-latest: true
+          node-version-file: ".nvmrc"
           cache: yarn
 
-      - name: Install dependencies
-        run: yarn install --frozen-lockfile
+      - name: 📥 Install dependencies
+        # even though this is called "npm-install" it does use yarn to install
+        # because we have a yarn.lock and caches efficiently.
+        uses: bahmutov/npm-install@v1
 
-      - name: Build
-        run: yarn build
-
-      - name: Test
-        run: yarn test & yarn size
+      - name: 🔐 Setup npm auth
+        run: |
+          echo "registry=https://registry.npmjs.org" >> ~/.npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> ~/.npmrc
 
-      - name: Publish
+        # This action has two responsibilities. The first time the workflow runs
+        # (initial push to a `release-*` branch) it will create a new branch and
+        # then open a PR with the related changes for the new version. After the
+        # PR is merged, the workflow will run again and this action will build +
+        # publish to npm.
+      - name: 🚀 PR / Publish
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          version: yarn run version
+          commit: "chore: Update version for release"
+          title: "chore: Update version for release"
+          publish: yarn release
```

Comment:
I think this also needs to `yarn build` - the prerelease yesterday is missing the `dist/` folders it looks like.  I wonder if that's also partially because our build goes to the root folder and not `packages/*`

Reply:
@chaance I might just be missing something but are we still missing a `yarn build` in the normal release flow?